### PR TITLE
@orta => Stitch in gravity and record artwork view using GravQL

### PR DESCRIPTION
**Pull request by mzikherman**
Relies on https://github.com/artsy/gravity/pull/11656 (I was testing locally all looking good).

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ const {
   GOOGLE_CSE_CX,
   GOOGLE_CSE_KEY,
   GRAVITY_API_BASE,
+  GRAVITY_GRAPHQL_ENDPOINT,
```
Comment (mzikherman) [inline src/config.js]: As opposed to `/api/v1`, GravQL is pseudo-under API V2: `/api/graphql`.

File: src/index.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ async function startApp() {
       })
       // Share with e.g. the Convection ApolloLink in mergedSchema.
       res.locals.dataLoaders = loaders // eslint-disable-line no-param-reassign
-
+      res.locals.accessToken = accessToken // eslint-disable-line no-param-reassign
```
Comment (mzikherman) [inline src/index.js]: Make this accessible to gravity merged schema

File: src/lib/mergeSchemas.js [hunk 3]
```diff
@@ -45,6 +45,29 @@ export function createConvectionLink() {
...
+  )
+
+  const authMiddleware = setContext((_request, context) => {
+    const locals = context.graphqlContext && context.graphqlContext.res.locals
+    const headers = { ...(locals && requestIDHeaders(locals.requestIDs)) }
+    Object.assign(headers, { "X-XAPP-TOKEN": config.GRAVITY_XAPP_TOKEN })
+    if (locals.accessToken) {
+      Object.assign(headers, { "X-ACCESS-TOKEN": locals.accessToken })
+    }
+    return { headers }
```
Comment (mzikherman) [inline src/lib/mergeSchemas.js]: pass along request id's, access token, xapp token

File: src/lib/mergeSchemas.js [hunk 4]
```diff
@@ -60,15 +83,28 @@ export async function mergeSchemas() {
...
       artist: Artist
     }
   `
 
+  // Add gravity schema first to prefer local MP schema.
+  // Add schemas after localSchema to prefer those over MP.
   const mergedSchema = _mergeSchemas({
-    schemas: [localSchema, convectionSchema, linkTypeDefs],
-    // Prefer others over the local MP schema.
+    schemas: [gravitySchema, localSchema, convectionSchema, linkTypeDefs],
```
Comment (mzikherman) [inline src/lib/mergeSchemas.js]: cc @damassi this is where we actually merge in gravity _first_, so as to keep preferring others further to the right, which we use in `onTypeConflict` below.

**Comment by dzucconi** [comment]
🎈Happy 1k 🎈

File: src/lib/mergeSchemas.js (hunk 4 above)
Comment (orta) [reply src/lib/mergeSchemas.js]: Makes sense 👍

**Comment by mzikherman** [comment]
This is probably mergeable? It's under the same `ENABLE_SCHEMA_STITCHING` flag for now, and the upstream endpoints/mutations all work identically with stitching enabled or not (exactly like Convection).

**Pull request merged**